crypto/x509/pcy_tree.c

```diff
@@ -110,6 +110,8 @@ static int tree_init(X509_POLICY_TREE **ptree, STACK_OF(X509) *certs,
 
     *ptree = NULL;
 
+    if (n < 0)
+        return X509_PCY_TREE_INTERNAL;
```

Comment:
FWIW, though the fix is correct as it stands, I would have instead redefined `n` to be the actual stack depth (without `-1`), and adjusted its use accordingly. That would have made for more robust (less obscure) code.